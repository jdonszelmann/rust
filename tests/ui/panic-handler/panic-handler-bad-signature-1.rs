//@ compile-flags:-C panic=abort

#![no_std]
#![no_main]

use core::panic::PanicInfo;

#[panic_handler]
fn panic(info: PanicInfo) -> () {}
//~^ ERROR function `panic` has a type that is incompatible with the declaration of `#[panic_handler]`
