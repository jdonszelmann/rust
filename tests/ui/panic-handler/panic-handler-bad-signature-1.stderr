error[E0053]: function `panic` has a type that is incompatible with the declaration of `#[panic_handler]`
  --> $DIR/panic-handler-bad-signature-1.rs:9:16
   |
LL | fn panic(info: PanicInfo) -> () {}
   |                ^^^^^^^^^ expected `&PanicInfo<'_>`, found `PanicInfo<'_>`
   |
note: expected this because of this attribute
  --> $DIR/panic-handler-bad-signature-1.rs:8:1
   |
LL | #[panic_handler]
   | ^^^^^^^^^^^^^^^^
   = note: expected signature `for<'a, 'b> fn(&'a PanicInfo<'b>) -> !`
              found signature `fn(PanicInfo<'_>) -> ()`
help: change the parameter type to match the declaration
   |
LL - fn panic(info: PanicInfo) -> () {}
LL + fn panic(info: &PanicInfo<'_>) -> () {}
   |

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0053`.
